### PR TITLE
Enable tensorflow-2.11 support for ROCm

### PR DESCRIPTION
**Pull request by srekolam**
This PR has the changes required to support tensorflow-2.11.0 version for ROCm.
I have used the local ROCm-5.5.1 installation and built the tensorflow-2.11.0-rocm-enhanced versions on a ubuntu-22.04 docker container having gcc@11.3.0
Attached is the packages.yaml (for reference) and the testcase result from the below -
spack load py-tensorflow@2.11.0-rocm-enhanced
python3 -c "from tensorflow.python.client import device_lib; print(device_lib.list_local_devices())"
spack unload
[tf-test-case-log.txt](
[packages_5.5.1-tf.yaml.txt](https://github.com/spack/spack/files/11833065/packages_5.5.1-tf.yaml.txt)
https://github.com/spack/spack/files/11833047/tf-test-case-log.txt)

### CODE REVIEW

**Comment by adamjstewart** [comment]
What's the status of upstreaming these changes so that we no longer need to patch TF for every release?

**Comment by srekolam** [comment]
@adamjstewart , here we are using the rocm's fork for tensorflow. Some of the changes from this fork might have already been available in the upstream project. From an earlier discussion on supporting spack's py-tensorflow package for rocm, we have agreed on using the rocm's fork for tensorflow.

**Review by adamjstewart — DISMISSED** [review]
I wonder if we should use a `url_for_version` so we don't have to repeat the URL every time.

**Comment by adamjstewart** [comment]
@spackbot run pipeline

**Comment by adamjstewart** [comment]
@spackbot run pipeline

**Comment by srekolam** [comment]
@spackbot rerun pipeline

**Comment by srekolam** [comment]
@spackbot rerun pipeline

**Comment by srekolam** [comment]
issue with mesa is resolved. need to run the pipeline again

@spackbot run pipeline

File: var/spack/repos/builtin/packages/mesa/package.py [hunk 1]
```diff
@@ -19,13 +19,16 @@ class Mesa(MesonPackage):
...
     version("main", tag="main")
-    version("22.3.2", sha256="c15df758a8795f53e57f2a228eb4593c22b16dffd9b38f83901f76cd9533140b")
-    version("22.2.5", sha256="850f063146f8ebb262aec04f666c2c1e5623f2a1987dda24e4361b17b912c73b")
+
     version(
-        "22.1.6",
-        sha256="22ced061eb9adab8ea35368246c1995c09723f3f71653cd5050c5cec376e671a",
+        "23.0.3",
+        sha256="386362a5d80df3b096636b67f340e1ce67b705b44767d5bdd11d2ed1037192d5",
         preferred=True,
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mesa/package.py]: The latest stable release is preferred by default, no need for `preferred=True`

Comment (srekolam) [reply var/spack/repos/builtin/packages/mesa/package.py]: thanks..made the changes.

**Review by adamjstewart — DISMISSED** [review]

**Comment by adamjstewart** [comment]
The error in CI looks like:
```
ERROR: ROCm version file not found in ['include/rocm-core/rocm_version.h', 'include/rocm_version.h']
```
So the build is actually broken.

**Comment by srekolam** [comment]
i do not see this issue and the build is passing..i just checked the py-tensorflow code and inside  the `third_party/gpus/find_rocm_config.py:70` , i see the below -
```python
def _find_rocm_config(rocm_install_path):
  def rocm_version_numbers(path):
    possible_version_files = [
        "include/rocm-core/rocm_version.h",  # ROCm 5.2
        "include/rocm_version.h",  # ROCm 5.1 and prior
    ]
```
in my local installation, i find the `rocm_version.h` at `/opt/rocm-5.5.1/include/rocm_version.h`.

**Comment by adamjstewart** [comment]
I don't think our build images have ROCm installed in a root directory, they are all installed by Spack. This is also why we currently don't test PyTorch + ROCm, because the build fails.

**Comment by adamjstewart** [comment]
#42851 has been merged, which disables TF+rocm in CI. Can you rebase this PR?

**Review by adamjstewart — APPROVED** [review]

**Comment by adamjstewart** [comment]
Really hoping to get this fork upstreamed and working with Spack-installed ROCm someday!

**Pull request merged**